pubsub/google/cloud/pubsub/connection.py

```diff
@@ -201,6 +202,7 @@ def topic_publish(self, topic_path, messages):
         :rtype: list of string
         :returns: list of opaque IDs for published messages.
         """
+        messages = _transform_messages_base64(messages, base64.b64encode)
```

Comment:
Each `message` in `messages` is mutable, so the transformation happens in place, so you don't need / want the return value.